sat/cli/status/status_module.py

```diff
@@ -560,16 +560,13 @@ def rows(self):
                     continue
 
                 try:
-                    template_name = component_session['template_name']
-                    component['Most Recent Session Template'] = bos_client.get_session_template(template_name)['name']
-                except APIError as err:
-                    LOGGER.warning('Could not retrieve BOS session template for component %s with session %s: %s',
-                                   component_xname, session_id, err)
-
+                    component['Most Recent Session Template'] = component_session['template_name']
                 except KeyError as err:
-                    LOGGER.warning('"%s" key missing from BOS response for component %s',
-                                   err, component_xname)
-                    continue
+                    raise StatusModuleException('"%s" key missing from BOS response for component %s',
+                                                err, component_xname) from err
+                    # LOGGER.warning('"%s" key missing from BOS response for component %s',
+                    #               err, component_xname)
+                    # continue
```

Comment:
It's preferred for this to remain a warning since it will just result in the "Most Recent Session Template" being "MISSING" for this component rather than failing for all of them because one is missing the session template in its session.

This is also where I thought it might be useful to include the BOS session ID in the log message. Something like:

```suggestion
                    LOGGER.warning('Unable to determine session template for %s due to missing '
                                   '"%s" key in BOS session %s.', component_xname, err, session_id)
                    continue
```

Also, in the future, please do not leave in commented out code. It can always be retrieved through git history if needed, and leaving it in just clutters the code.